### PR TITLE
Adds configurable target coordinate frame for teleop output poses

### PR DESCRIPTION
**Pull request by rwiltz**
# Description

Add a target_T_world parameter to IsaacTeleopDevice.advance() and TeleopSessionLifecycle.step() that left-multiplies the anchor matrix to rebase all output poses into an arbitrary target coordinate frame (e.g. robot base link for IK).
Add a config-driven alternative via IsaacTeleopCfg.target_frame_prim_path: when set to a USD prim path, the device automatically reads the prim's world transform each frame via USDRT/Fabric and uses its inverse as the rebase matrix, removing the need for callers to compute it manually.
Add _to_numpy_4x4 helper that accepts np.ndarray, torch.Tensor, wp.array, or any duck-typed .numpy() object and normalizes to float32 NumPy.
Add comprehensive unit tests for the conversion helper, matrix rebase math, and config-vs-explicit auto-selection logic.

Fixes # (issue)

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Comment by rwiltz** [comment]
@greptile-apps

File: source/isaaclab_teleop/isaaclab_teleop/session_lifecycle.py [hunk 1]
```diff
@@ -21,8 +21,52 @@
...
+    Args:
+        mat: A (4, 4) transform matrix.
+
+    Returns:
+        A (4, 4) float32 :class:`numpy.ndarray`.
+    """
+    if isinstance(mat, np.ndarray):
+        return np.asarray(mat, dtype=np.float32)
+    if isinstance(mat, torch.Tensor):
+        return np.asarray(np.from_dlpack(mat.detach().cpu()), dtype=np.float32)
```
[the remark this answers is not in the document]
Comment (rwiltz) [reply source/isaaclab_teleop/isaaclab_teleop/session_lifecycle.py:57]: Safe to ignore. The core isaaclab package pins numpy>=2 (source/isaaclab/setup.py), and isaaclab_teleop is always installed alongside it. Since np.from_dlpack has been available since NumPy 1.22, the floor is nearly a full major version above what's needed — there is no supported environment where this call could raise AttributeError. Adding a hasattr guard would be unreachable dead code.

File: source/isaaclab_teleop/isaaclab_teleop/session_lifecycle.py [hunk 2]
```diff
@@ -360,11 +416,12 @@ def step(self, anchor_world_matrix_fn: Callable[[], np.ndarray] | None = None) -
...
-        action_np = result["action"][0]
-
-        # Convert to torch tensor and move to device
-        action = torch.from_numpy(np.asarray(action_np, dtype=np.float32)).to(self._device)
+        # Extract the flattened action array (DLPack-compatible) from
+        # TensorReorderer and move to the simulation device.
+        action_array = result["action"][0]
+        action = torch.from_dlpack(action_array).to(  # type: ignore[attr-defined]
+            dtype=torch.float32, device=self._device
+        )
```
[the remark this answers is not in the document]
Comment (rwiltz) [reply source/isaaclab_teleop/isaaclab_teleop/session_lifecycle.py:425]: Safe to ignore. TensorReorderer._compute_fn produces a plain np.float32 ndarray (np.zeros(..., dtype=np.float32)), and NumPy arrays fully implement __dlpack__/__dlpack_device__ on our pinned numpy>=2. torch.from_dlpack on a NumPy array is a well-supported zero-copy interop path. The suggested try/except fallback would be dead code.

**Comment by rwiltz** [comment]
@greptile-apps

**Comment by rwiltz** [comment]
@greptile-apps

**Comment by rwiltz** [comment]
@greptile-apps

**Comment by rwiltz** [comment]
@greptile-apps

**Comment by qianl-nv** [comment]
Change lgtm from Arena side. I've tested the Arena locomanipulaiton teleop/record/replay is working correctly with this MR.

**Review by hougantc-nvda — APPROVED** [review]
LGTM. 
Only question for future discussion is if we need to support USD (non usdrt path) in the future
